docs/docs/troubleshooting.md

```diff
@@ -1,5 +1,175 @@
+ - Use a DNS provider other than CloudFlare
+ - Add system packages to the user's (jupyterlab) image
+ - Provide individual users with unique environments and cloud instance types
+
+## Behavior
+
+### DNS domain=`your_nebari_domain` record does not exist
+
+During your initial Nebari deployment, at the end of the `04-kubernetes-ingress` stage, you may receive an output message stating that the DNS record for `your_nebari_domain` "appears not to exist, has recently been updated, or has yet to fully propagate."
+
+As the output message mentions, this is likely the result of the non-deterministic behavior of DNS.
+
+Without going into a deep dive of what DNS is or how it works, the issue encountered here is that when Nebari tries to look up the IP address associated with the DNS record, `your_nebari_domain`, nothing is returned. Unfortunately, this "lookup" is not as straightforward as it sounds. To lookup the correct IP associated with this domain, many intermediate servers (root, top level domain, and authoritative nameservers) are checked, each with their own cache, and each cache has its own update schedule (usually on the order of minutes, but not always).
+
+(If you are so inclined to learn more about DNS, [see this interesting comic](https://howdns.works/))
+
+As the output message mentions, it will ask if you want it to retry this DNS lookup again after another wait period; this wait period keeps increasing after each retry. However, it's possible that after waiting 15 or more minutes that the DNS still won't resolve.
+
+At this point, feel free to cancel the deployment and rerun the same deployment command again in an hour or two. Although not guaranteed, it's extremely likely that the DNS will resolve correctly after this prolonged wait period.
+
+### User instances on AWS occasionally die ~30 minutes after spinning up a large dask cluster
+
+AWS uses Elastic Kubernetes Service for hosting the Kubernetes cluster. [It requires the use of at least two availability zones](https://docs.aws.amazon.com/eks/latest/userguide/infrastructure-security.html). The Nebari cluster has an autoscaler with a default service that automatically balances the number of EC2 instances between the two availability zones. When large dask clusters are initialized and destroyed, the autoscaler attempts to reschedule a user pod. This reschedule operation occurs in the other availability zone. Unfortunately, Kubernetes doesn't successfully transfer the active pod to the other zone, and the pod dies.
+
+To stop this from happening, the autoscaler service "AZRebalance" needs to be manually suspended. You can [follow the steps in the AWS documentation to suspend the AZRebalance service](https://docs.aws.amazon.com/autoscaling/ec2/userguide/as-suspend-resume-processes.html).
+
+### Why is the `NEBARI_KUBECONFIG` file in `/tmp`?
+
+Nebari regenerates this file on every run. Yes, it will be trashed by the operating system's tmpdir cleanup, but it will be repopulated.
```

Comment:
that's correct, though I would prefer to present more details on how it can be generated as a reference:
```
Since the `KUBECONFIG_FILE` is stored in `/temp`, the system will remove that file during its clean-up process. If you want to re-generate the file again, you can run the `qhub deploy` command again as Nebari will update/create a `KUBECONFIG_FILE` in for you for each deployment.
```

Reply:
Done.